libsupport/include/katana/HostAllocator.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdlib>
+#include <cstring>
+
+#include "katana/config.h"
+
+namespace katana {
+
+enum class MemoryPinType { Swappable = 0, Pinned = 1 };
+
+class BaseHostAllocator {
+public:
+  virtual void* Malloc(const size_t n_bytes) const = 0;
+  virtual void* Calloc(const size_t n_items, const size_t item_size) const = 0;
+  virtual void* Realloc(void* ptr, const size_t n_bytes) const = 0;
+  virtual void Free(void* ptr) const = 0;
+  virtual MemoryPinType GetType() const = 0;
```

Comment:
As mentioned in the previous comment, I think we should abstract this into the idea of slow and fast allocators to avoid having tying performance considerations to pinning which isn't directly related. A pinned allocator that used a large pool of pinned memory could be fast, and allocators which do not pin could still be slow if they are designed to reduce wasted memory at all costs.

What do you think?

Reply:
I've changed to `IsFastAlloc()` method, will push soon.